modules/packages/ExternalArray.chpl

```diff
@@ -25,6 +25,19 @@
 module ExternalArray {
   use ChapelStandard;
 
+  extern type chpl_free_func;
+  private extern const CHPL_FREE_FUNC_NIL: chpl_free_func;
+
+  extern record chpl_external_array {
+    var elts: c_void_ptr;
+    var size: uint;
+
+    var freer: chpl_free_func;
+  }
+  extern proc chpl_make_external_array(elt_size: uint, num_elts: uint):
+    chpl_external_array;
+  extern proc chpl_call_free(x: chpl_external_array);
```

Comment:
`chpl_call_free` is pretty nonspecific.  Can we name this `chpl_free_external_array()`, by analogy with the `make` function?  (It's okay that the actual free operation is indirect, that's true for a long way down the call stack.)